tdo.sh

```diff
@@ -43,6 +43,27 @@ For more information, visit https://github.com/2kabhishek/tdo
 EOF
 }
 
+# validate whether a variable is set to true or false, and set to default otherwise
+validate_and_set() {
+    local default_val="${2:-true}"
+    local var_value="${1:-$default_val}"
+    if [[ "$var_value" != "true" && "$var_value" != "false" ]]; then
+        var_value=$default_val
+    fi
+    echo "$var_value"
+}
+
+# setup config variables
+config_setup() {
+  source $HOME/.config/tdorc
+
+  add_entry_timestamp="$(validate_and_set "${ADD_ENTRY_TIMESTAMP}" "true")"
+  add_new_note_timestamp="$(validate_and_set "${ADD_NEWNOTE_TIMESTAMP}" "false")"
+  filename_as_title="$(validate_and_set "${FILE_NAME_AS_TITLE}" "false")"
+  entry_timestamp_format="${ENTRY_TIMESTAMP_FORMAT:-"## %a, %I:%M %p"}"
+  note_timestamp_format="${NOTE_TIMESTAMP_FORMAT:-"## %a. %b %d, %Y - %I:%M %p"}"
+}
```

Comment:
this can be simplified, the other function won't be necessary
```suggestion
config_setup() {
  source "$HOME/.config/tdorc"

  add_entry_timestamp="${ADD_ENTRY_TIMESTAMP:-true}"
  add_new_note_timestamp="${ADD_NEWNOTE_TIMESTAMP:-false}"
  filename_as_title="${FILE_NAME_AS_TITLE:-false}"
  entry_timestamp_format="${ENTRY_TIMESTAMP_FORMAT:-"## %a, %I:%M %p"}"
  note_timestamp_format="${NOTE_TIMESTAMP_FORMAT:-"## %a. %b %d, %Y - %I:%M %p"}"
}
```